instrumentation/kubernetes/src/main/java/io/opentelemetry/auto/instrumentation/kubernetes/KubernetesClientInstrumentation.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.auto.instrumentation.kubernetes;
+
+import static java.util.Collections.singletonMap;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+
+import com.google.auto.service.AutoService;
+import io.kubernetes.client.openapi.ApiClient;
+import io.opentelemetry.auto.tooling.Instrumenter;
+import java.util.List;
+import java.util.Map;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import net.bytebuddy.matcher.ElementMatchers;
+import okhttp3.Interceptor;
+
+/**
+ * @author zuoxiu.jm
+ * @version : KubernetesClientInstrumentation.java, v 0.1 2020年07月08日 4:35 PM zuoxiu.jm Exp $
+ */
+@AutoService(Instrumenter.class)
+public class KubernetesClientInstrumentation extends Instrumenter.Default {
+
+  public KubernetesClientInstrumentation() {
+    super("kubernetes");
+  }
+
+  @Override
+  public ElementMatcher<TypeDescription> typeMatcher() {
+    return named("io.kubernetes.client.openapi.ApiClient");
+  }
+
+  @Override
+  public String[] helperClassNames() {
+    return new String[] {
+      packageName + ".KubernetesClientDecorator",
+      packageName + ".TracingInterceptor",
+      packageName + ".KubernetesRequestDigest",
+      packageName + ".KubernetesResource",
+      packageName + ".KubernetesVerb",
+      packageName + ".ParseKubernetesResourceException",
+    };
+  }
+
+  @Override
+  public Map<? extends ElementMatcher<? super MethodDescription>, String> transformers() {
+    return singletonMap(
+        ElementMatchers.isMethod()
+            .and(named("initHttpClient"))
+            .and(ElementMatchers.takesArguments(1))
+            .and(ElementMatchers.takesArgument(0, named("java.util.List"))),
+        KubernetesClientInstrumentation.class.getName() + "$KubernetesAdvice");
+  }
+
+  public static class KubernetesAdvice {
+    @Advice.OnMethodExit(suppress = Throwable.class)
+    public static void addTracingInterceptor(
+        @Advice.This ApiClient apiClient,
+        @Advice.Argument(0) final List<Interceptor> interceptors) {
+
+      for (final Interceptor interceptor : interceptors) {
+        if (interceptor instanceof TracingInterceptor) {
+          return;
+        }
+      }
+
+      apiClient.setHttpClient(
```

Comment:
Can you modify incoming parameter `interceptors` and add a new one there? I don't know this API, so don't have any hard preferences :)

Reply:
the `interceptors` are immutable so i have to craft a new list instead of updating in-place..

https://github.com/kubernetes-client/java/blob/d5b80572498122bb45c830c74f89b871dbbdfc92/kubernetes/src/main/java/io/kubernetes/client/openapi/ApiClient.java#L93

Reply:
You can change this method to `Advice.OnMethodEnter`, change `interceptors` argument to `readOnly = false` and construct new list which includes both old value of `interceptors` and `TracingInterceptor`